### PR TITLE
Testing error handling for Syndeo and SQL transform

### PR DESCRIPTION
**Pull request by pabloem**
The new test added in this change depends on https://github.com/apache/beam/pull/25465 to work.

### CODE REVIEW

**Comment by pabloem** [comment]
r: @manavgarg - a test that exercises the initial part of DLQ for SQL Transforms : )

**Comment by pabloem** [comment]
unit tests are failing pending for the Beam snapshot releases to come out

File: syndeo-template/src/main/java/com/google/cloud/syndeo/common/ProviderUtil.java [hunk 1]
```diff
@@ -136,6 +133,17 @@ public static PCollectionRowTuple applyConfigs(
       }
 
       tuple = tuple.apply(spec.inputId, transform.buildTransform());
+
+      if (tuple.getAll().containsKey(ERRORS_TAG)) {
+        // TODO(pabloem): Deal with errors.
+        PCollectionRowTuple.of("input", tuple.get(ERRORS_TAG))
```
Comment (manavgarg) [inline syndeo-template/src/main/java/com/google/cloud/syndeo/common/ProviderUtil.java:139]: so right now we are just incrementing the metric to detect the error right.

**Review by manavgarg — APPROVED** [review]
LGTM, thanks.

File: syndeo-template/src/main/java/com/google/cloud/syndeo/common/ProviderUtil.java (hunk 1 above)
Comment (pabloem) [reply syndeo-template/src/main/java/com/google/cloud/syndeo/common/ProviderUtil.java:139]: that's correct Manav! We can figure out how to act on that afterwards

**Comment by pabloem** [comment]
failure in Load TEsts due to resource scarcity / flakiness

**Pull request merged**